CS3219-AY2223S1/cs3219-project-ay2223s1-g12
# Dockerize application

Fixes #134 
# Summary of changes made:
- Added a Dockerfile (and `.dockerignore` where applicable) for each microservice
- Added a `docker-compose.yml` file to build the docker containers 
- Updated URLs (specifically, api call from `matching-service` to `question-service`) to be recognisable by Docker 
- Updated README.md with instructions on how to deploy the application locally using Docker

## To test: 

1. Ensure that Docker desktop application is running. 
2. Go to the root directory of the repository.
3. Run the following command:
    ```bash
    #build and run each microservice using docker-compose
    docker-compose up --build -d
    ```
4. After the docker containers are up, navigate to  `localhost:3000` on your web browser and start using the application. 


#### Tear down:
1. Run the following command:
   ```bash
   # tear down containers 
   docker-compose down
   ``` 

## Key changes to take note of:
In Docker, when calling a service from another service, we have to specify the service name in the URL e.g `http://question-service:8002/api` instead of `http://localhost:8002/api`

<img width="1168" alt="image" src="https://user-images.githubusercontent.com/77088875/200271948-bb921590-dc12-4247-9e31-8d35a1baa128.png">

<img width="596" alt="image" src="https://user-images.githubusercontent.com/77088875/200271987-0f63047b-5a18-41fb-b211-02e525cb45e4.png">

hence i set a toggle `isDockerized` (as shown above). When running on local without docker, we need to change the toggle to `false` so that the correct endpoint will be called.
